main.sh

```diff
@@ -266,6 +266,59 @@ function deploy() {
 	dep deploy "$GITHUB_BRANCH"
 }
 
+function block_emails() {
+
+	hosts_block_email=$(shyaml get-value "$GITHUB_BRANCH.block_emails" < "$hosts_file" 2>/dev/null || exit 0)
+
+	if [[ -n "$hosts_block_email" ]]; then
+		BLOCK_EMAILS="$hosts_block_email"
+	fi
+
+	if [[ -n "$BLOCK_EMAILS" ]]; then
+
+		# priority: 1. hosts.yml 2. vip 3. WP
+		echo -e "\033[34mSETUP EMAIL BLOCKING\033[0m"
+		if [[ -d "$HTDOCS/wp-content/client-mu-plugins" ]]; then
+		   BLOCK_EMAIL_DIR="$HTDOCS/wp-content/client-mu-plugins"
+		elif [[ -d "$HTDOCS/wp-content/mu-plugins" ]]; then
+		   BLOCK_EMAIL_DIR="$HTDOCS/wp-content/mu-plugins"
+		fi
+
+		hosts_block_email_dir=$(shyaml get-value "$GITHUB_BRANCH.block_emails_plugin_path" < "$hosts_file" 2>/dev/null || exit 0)
+
+		if [[ -n "$hosts_block_email_dir" ]]; then
+			BLOCK_EMAIL_DIR="$HTDOCS/wp-content/$hosts_block_email_dir"
+		elif [[ -n "$BLOCK_EMAILS_PLUGIN_PATH" ]]; then
+			BLOCK_EMAIL_DIR="$HTDOCS/wp-content/$BLOCK_EMAILS_PLUGIN_PATH"
+		fi
+
+		BLOCK_EMAIL_DIR="${BLOCK_EMAIL_DIR%/}"
+
+		# using this naming convention by default to load this plugin first in mu-plugin loading phase.
+		BLOCK_EMAIL_PLUGIN_NAME="000-block-emails.php"
+
+		hosts_block_email_file_name=$(shyaml get-value "$GITHUB_BRANCH.block_emails_plugin_file_name" < "$hosts_file" 2>/dev/null || exit 0)
+
+		if [[ -n "$hosts_block_email_file_name" ]]; then
+			BLOCK_EMAIL_PLUGIN_NAME="${hosts_block_email_file_name}.php"
+		elif [[ -n "$BLOCK_EMAILS_PLUGIN_FILE_NAME" ]]; then
```

Comment:
Remove the workflow file env variables option. Keep only the option to configure this from `hosts.yml` so that the person setting this up is well aware of which branch this email block functionality is getting set up in and does not accidentally set it up for all branches.